dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -58,10 +58,18 @@ NvJpeg2000DecoderInstance::NvJpeg2000DecoderInstance(
   nvjpeg2k_handle_ = NvJpeg2kHandle(&nvjpeg2k_dev_alloc_, &nvjpeg2k_pin_alloc_);
   DALI_ENFORCE(nvjpeg2k_handle_, "NvJpeg2kHandle initalization failed");
 
+  bool resource_allocation_successful = true;
   ForEachThread(*tp_, [&](int tid) noexcept {
-    CUDA_CALL(cudaSetDevice(device_id));
-    per_thread_resources_[tid] = {nvjpeg2k_handle_, device_memory_padding, device_id_};
+    try {
+      CUDA_CALL(cudaSetDevice(device_id));
+      per_thread_resources_[tid] = {nvjpeg2k_handle_, device_memory_padding, device_id_};
+    } catch (dali::CUDAError&) {
+      resource_allocation_successful = false;
+    }
```

Comment:
This is just wrong. If you want to fix the problem, not just silence the tool, then it should be fixed at `ForEachThread`, not here (here you could just remove the noexcept).
Also, this piece of code could still throw a plain std::bad_alloc.